### PR TITLE
File upload

### PR DESCRIPTION
**Pull request by velrest**

### CODE REVIEW

**Comment by velrest** [comment]
Depends on #1

**Comment by czosel** [comment]
Conflicts ...

File: addon/components/document-upload-button.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            document: documentModel,
+          });
+          await fileModel.save();
+
+          const response = await fetch(fileModel.uploadUrl, {
+            method: "put",
+            body: file,
+          });
+
+          console.error("Needs error handling here", response);
```
Comment (czosel) [inline addon/components/document-upload-button.js:40]: Old?

File: app/styles/_skeleton-loading.scss [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+// forked from https://codepen.io/viktorstrate/pen/yoBRLy
+@mixin background-gradient {
+  background-image: linear-gradient(
+    90deg,
+    $base-color 0px,
+    $shine-color 40px,
+    $base-color 80px
+  );
+  background-size: 100vw;
+}
```
Comment (czosel) [inline app/styles/_skeleton-loading.scss:27]: double?

File: translations/de.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  download: "Herunterladen"
+  delete: "Löschen"
+  upload-file: "Datei hochladen"
+
+  errors:
+    save-file: "Beim Herunterladen der Datei ist ein Fehler aufgetreten. Bitte versuchen Sie es erneut."
+    delete-document: "Beim Löschen des Dokummentes ist ein Fehler aufgetreten. Bitte versuchen Sie es erneut."
+    fetch-categories: "Beim laden der Kategorien ist ein Fehler aufgetreten."
+    upload-document: |-
+      Beim hochladen {count, plural,
```
Comment (czosel) [inline translations/de.yaml:13]:
```suggestion
      Beim Hochladen {count, plural,
```

File: translations/de.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  loading: "Lädt..."
+  nothing-found: "Wir haben nichts gefunden..."
+  download: "Herunterladen"
+  delete: "Löschen"
+  upload-file: "Datei hochladen"
+
+  errors:
+    save-file: "Beim Herunterladen der Datei ist ein Fehler aufgetreten. Bitte versuchen Sie es erneut."
+    delete-document: "Beim Löschen des Dokummentes ist ein Fehler aufgetreten. Bitte versuchen Sie es erneut."
+    fetch-categories: "Beim laden der Kategorien ist ein Fehler aufgetreten."
```
Comment (czosel) [inline translations/de.yaml:11]:
```suggestion
    fetch-categories: "Beim Laden der Kategorien ist ein Fehler aufgetreten."
```

**Review by czosel — APPROVED** [review]
:+1:

**Pull request merged**